src/libxrpl/protocol/Serializer.cpp

```diff
@@ -71,6 +72,32 @@ Serializer::add64(std::uint64_t i)
     return ret;
 }
 
+int
+Serializer::addi32(std::int32_t i)
+{
+    int ret = mData.size();
+    static constexpr std::size_t SIZE =
+        sizeof(std::int32_t) / sizeof(unsigned char);
+    static_assert(SIZE == 4);
+    mData.resize(ret + SIZE);
+    unsigned char* p = &mData.back() + 1 - SIZE;
+    boost::endian::store_big_s32(p, i);
```

Comment:
Why not use the same serialization approach as in other methods? I.e. `push_back/shift`?

Reply:
Because the hour and time ratio are not the same.